core/domain/base_model_validators.py

```diff
@@ -33,6 +33,10 @@
 
 USER_ID_REGEX = 'uid_[a-z]{32}'
 
+# These are model ids used for bots like suggestion bot and migration
+# bot. We do not need to perform any validation checks for such models.
```

Comment:
Actually, thinking about it, I don't think I agree with this. It's true that bot models may have their own set of constraints that are a bit different from regular models. But I don't think it's right to say that a bot model can literally contain anything.

I think that each model whose ID may be a bot name should recognize that and do the necessary checks based on the item.id, but I don't think there should be a blanket exclusion. Some (probably most?) validation checks ought to hold for bot models too, and the overall aim of this project is to fully validate all data on the server, which includes the models that are associated with bots.

(Also, it may be that, for certain classes, there _shouldn't_ be models associated with bots, and that the existence of those models is a mistake. We should probably check for that too.)

Reply:
Makes sense. 

Bot models are also user models, so basically I have updated user id regex to consider the two bot ids as a valid pattern and they will be validated like other user models.